tests/integration/test_rootcheck/test_configuration/test_rootcheck.py

```diff
@@ -0,0 +1,238 @@
+                 table request.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - get_configuration:
+            type: fixture
+            brief: Get configurations from the module.
+        - configure_environment:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - restart_service:
+            type: fixture
+            brief: restart the services
+        - clean_alert_logs:
+            - type: fixture
+            - brief: reset the content of the alert logs
+    assertions:
+        - Verify that rootcheck events are added into the database
+        - Verify that the rootcheck events are updated on the database
+        - Verify that the rootcheck events are deletet from the database
+    input_description: Different test cases are contained in an external YAML file (wazuh_manager_conf.yaml)
+                       which includes configuration settings for the 'rootcheck' module.
+    expected_output:
+        - r'.*not found in Database'
+        - r'.*not found in alerts file'
+        - r'.*not found in Database'
+        - First time in log was updated after insertion
+        - Updated time in log was not updated
+        - Wazuh DB returned an error trying to delete the agent
+        - Rootcheck events were not deleted
+
+    '''
+    agents = create_agents(test_metadata["agents_number"], SERVER_ADDRESS, CRYPTO)
+
+    for agent in agents:
+        agent.modules['rootcheck']['status'] = 'enabled'
+
+    injectors = create_injectors(agents)
+
+    # Let rootcheck events to be sent for 60 seconds
+    time.sleep(60)
+
+    for injector in injectors:
+        injector.stop_receive()
+
+    # Service needs to be stopped otherwise db lock will be held by Wazuh db
+    control_service('stop')
+
+    # Check that logs have been added to the sql database
+    for agent in agents:
+        rows = retrieve_rootcheck_rows(agent.id)
+        db_string = [row[3] for row in rows]
+        logs_string = [':'.join(x.split(':')[2:]) for x in
+                      agent.rootcheck.messages_list]
+
+        alerts_description = []
+        with open(ALERTS_JSON_PATH, 'r') as f:
+            for line in f:
+                if '"decoder":{"name":"rootcheck"}' in line:
+                    try:
+                        json_lines = json.loads(line.strip())
+                        alerts_description.append(json_lines['full_log'])
+                    except json.JSONDecodeError:
+                        print(f"Invalid json {line.strip()}")
+
+        for log in logs_string:
+            assert log in db_string, f"Log: \"{log}\" not found in Database"
+            if log not in ['Starting rootcheck scan.',
+                           'Ending rootcheck scan.']:
+                assert log in alerts_description, f"Log: \"{log}\" " \
+                                                  "not found in alerts file"
+
+    if test_metadata["check_updates"]:
```

Comment:
Maybe this can be a new test case? I see very complicated this test to put all together.